src/words/fast.jl

```diff
@@ -0,0 +1,110 @@
+# TODO: the input + idx combination should probably be replaced by a stream
```

Comment:
yes, and then the ToenBuffer can stop being mutable itself,
right?

Reply:
Yeah, and this would be nice for streaming as well. This requires some kind of buffered reader that can do lookaheads though, so I figure we should build that infrastructure separately.

Reply:
Fair, so we can merge without waiting for that